### DISCUSSION TITLE
Is it possible to ask an update using MQTT?

### DISCUSSION THREAD

**Discussion by ektoekto** [tillsteinbach/WeConnect-mqtt#351] [General]
All messages that I get trough MQTT is pushed to an array of objects. So I have all the data available, messages just update or create new objects. I noticed today that some of the objects ware not updating. For example the ChargingStatus had received last update three hours ago and it didn't include all the data.
![kuva](https://github.com/user-attachments/assets/16a2d148-46f8-43fc-8a29-b5d42195e77f)

I noticed on my lunch brake that the car was charging, but my system said that it should not be charging, and should start later on. After investigating I noticed that the chargingState did not include anything.

Is there a way to force an update? Something I could send out trough MQTT that would cause the system to update it self?

**Comment by Thyraz** [comment]
You can force the vehicle to wake up and send updates to the cloud e.g. by changing the AC temperature to +2° and a few seconds back to the old value (without the need to activate the AC).
Or change the battery charing limit to something else and then back.

Changing these values always woke the car and made it talking the cloud in my case.

You can also do a one-time update to receive values from the cloud to MQTT faster than the normal 300s cycle after that using the above "trick" by sending _True_ to _weconnect/0/mqtt/weconnectForceUpdate_writetopic_.
(But give the vehicle some time to wake up and sync to the cloud.)

But the vehicle will not update always all values to the cloud when it syncs.
Sometimes some values and the corresponding carCapturedTimestamps stay the same. 
So not sure if this will fix problems with wrong / old states in the cloud.

Just keep in mind that there's a daily rate limit where all the requests count against (also the official app  and the 300s automatic requests of Weconnect-mqtt).
But in my experience the 300s are quite conservative and there's some room for extra calls when needed.

**Reply by tillsteinbach** [reply]
On some cars there is also a wakeup topic in the controls
